core/__tests__/AsyncStorage.test.ts

```diff
@@ -0,0 +1,121 @@
+import AsyncStorage from '../src/AsyncStorage';
+
+class StorageMock implements IStorageBackend<any> {
+  getSingle = jest.fn();
+  setSingle = jest.fn();
+  getMany = jest.fn();
+  setMany = jest.fn();
+  removeSingle = jest.fn();
+  removeMany = jest.fn();
+  getKeys = jest.fn();
+  dropStorage = jest.fn();
+}
+
+describe('AsyncStorage', () => {
+  const mockedStorage = new StorageMock();
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('main API', () => {
+    it('handles basic set/read/remove calls', async () => {
```

Comment:
I'd use [test.each](https://jestjs.io/docs/en/api#testeachtable-name-fn-timeout) for these tests for convenience